oak-blob-cloud/src/main/java/org/apache/jackrabbit/oak/blob/cloud/s3/S3RequestDecorator.java

```diff
@@ -58,20 +88,63 @@ public PutObjectRequest decorate(PutObjectRequest request) {
      * Set encryption in {@link CopyObjectRequest}
      */
     public CopyObjectRequest decorate(CopyObjectRequest request) {
+        ObjectMetadata metadata;
         switch (getDataEncryption()) {
             case SSE_S3:
-                ObjectMetadata metadata = request.getNewObjectMetadata() == null
+                metadata = request.getNewObjectMetadata() == null
                                 ? new ObjectMetadata()
                                 : request.getNewObjectMetadata();
                 metadata.setSSEAlgorithm(ObjectMetadata.AES_256_SERVER_SIDE_ENCRYPTION);
                 request.setNewObjectMetadata(metadata);
                 break;
+            case SSE_KMS:
```

Comment:
As above optimize block to remove redundancies. Also, better to be consistent i.e. either use switch case or if/else everywhere